intake_stac/catalog.py

```diff
@@ -230,6 +283,20 @@ class StacItem(AbstractStacCatalog):
     name = 'stac_item'
     _stac_cls = pystac.Item
 
+    def __getitem__(self, key):
+        result = super().__getitem__(key)
+        # TODO: handle non-string assets?
```

Comment:
i haven't come across this in the wild. are they always strings? here for example I see asset["0"] https://cmr.earthdata.nasa.gov/stac/NSIDC_ECS/collections/NSIDC-0723.v4/items